internal/api/errors.go

```diff
@@ -0,0 +1,222 @@
+
+	var dbResources []db.Resource
+	_, err := h.DB.Select(&dbResources,
+		`SELECT * FROM resources WHERE scrape_error_message != '' ORDER BY id`)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+
+	var result struct {
+		ResourceScrapeErrors []ResourceScrapeError `json:"resource_scrape_errors"`
+	}
+	for _, res := range dbResources {
+		projectID := ""
+		// .ScopeUUID is either a domain- or project UUID.
+		if res.ScopeUUID != res.DomainUUID {
+			projectID = res.ScopeUUID
+		}
+
+		result.ResourceScrapeErrors = append(result.ResourceScrapeErrors,
+			ResourceScrapeError{
+				ProjectUUID: projectID,
+				DomainUUID:  res.DomainUUID,
+				AssetType:   string(res.AssetType),
+				Checked: Checked{
+					AtUnix:       res.CheckedAt.Unix(),
+					ErrorMessage: res.ScrapeErrorMessage,
+				},
+			})
+	}
+
+	respondwith.JSON(w, http.StatusOK, result)
+}
+
+func (h handler) GetAssetScrapeErrors(w http.ResponseWriter, r *http.Request) {
+	sre.IdentifyEndpoint(r, "/v1/admin/asset-scrape-errors")
+	_, token := h.CheckToken(w, r)
+	if token == nil {
+		return
+	}
+	if !token.Require(w, "cluster:access") {
+		return
+	}
+
+	var result struct {
+		AssetScrapeErrors []AssetError `json:"asset_scrape_errors"`
+	}
+
+	var dbResources []db.Resource
+	_, err := h.DB.Select(&dbResources,
+		`SELECT * FROM resources ORDER BY id`)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+
+	for _, res := range dbResources {
+		var dbAssets []db.Asset
+		_, err := h.DB.Select(&dbAssets,
+			`SELECT * FROM assets
+			 WHERE scrape_error_message != '' AND resource_id = $1
+			 ORDER BY id
+			`, res.ID)
+		if respondwith.ErrorText(w, err) {
+			return
+		}
+
+		projectID := ""
+		// res.ScopeUUID is either a domain- or project UUID.
+		if res.ScopeUUID != res.DomainUUID {
+			projectID = res.ScopeUUID
+		}
+
+		for _, a := range dbAssets {
+			result.AssetScrapeErrors = append(result.AssetScrapeErrors,
+				AssetError{
+					AssetUUID:   a.UUID,
+					ProjectUUID: projectID,
+					DomainUUID:  res.DomainUUID,
+					AssetType:   string(res.AssetType),
+					Checked: &Checked{
+						AtUnix:       a.CheckedAt.Unix(),
+						ErrorMessage: a.ScrapeErrorMessage,
+					},
+				})
+		}
+	}
+
+	respondwith.JSON(w, http.StatusOK, result)
+}
+
+func (h handler) GetAssetResizeErrors(w http.ResponseWriter, r *http.Request) {
+	sre.IdentifyEndpoint(r, "/v1/admin/asset-resize-errors")
+	_, token := h.CheckToken(w, r)
+	if token == nil {
+		return
+	}
+	if !token.Require(w, "cluster:access") {
+		return
+	}
+
+	var result struct {
+		AssetResizeErrors []AssetError `json:"asset_resize_errors"`
+	}
+
+	var dbResources []db.Resource
+	_, err := h.DB.Select(&dbResources,
+		`SELECT * FROM resources ORDER BY id`)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+
+	for _, res := range dbResources {
+		var ops []db.FinishedOperation
+		_, err := h.DB.Select(&ops, `
+			SELECT o.* FROM finished_operations o
+				JOIN assets a ON a.id = o.asset_id
+			 WHERE a.resource_id = $1
+		`, res.ID)
```

Comment:
We only care about assets that are still problematic. So we want to skip "errored" operations where a more recent operation on the same asset finished as "succeeded" or "cancelled" or "failed". As far as I can see, there is nothing here that does that. I was going to suggest you look at `h.GetRecentlyFailedOperations` to see how it's done there, but I remembered that I read about a feature in Postgres that would let us express this in SQL: https://www.yogeshchauhan.com/167/postgres/the-confusing-unique-and-useful-feature-in-postgres-distinct-on

I tested a bit on the eu-de-1 DB and came up with this query:
```sql
WITH latest_finished_operations AS (
  SELECT DISTINCT ON (asset_id) o.* FROM finished_operations o
    JOIN assets a ON a.id = o.asset_id
    WHERE a.resource_id = $1
  ORDER BY o.asset_id, o.finished_at DESC)
SELECT * FROM latest_finished_operations WHERE outcome = 'errored'
```

Please have a look if this works here, and also add a testcase where an asset has an `errored` operation that does not show up in this endpoint because of a more recent `succeeded` operation.